### PR TITLE
[Feature] Change Mana Costs to use Signed Int

### PR DESCRIPTION
**Pull request by Aeadoin**
Changed Spell Mana Costs from Unsigned Int16 to Signed Int32.

### CODE REVIEW

File: common/spdat.h [hunk 1]
```diff
@@ -1298,7 +1298,7 @@ struct SPDat_Spell_Struct
 /* 016 */	uint32 buff_duration_formula; // -- DURATIONBASE
 /* 017 */	uint32 buff_duration; // -- DURATIONCAP
 /* 018 */	uint32 aoe_duration;	// sentinel, rain of something -- IMPACTDURATION
-/* 019 */	uint16 mana; // Mana Used -- MANACOST
```
Comment (Kinglykrab) [inline common/spdat.h:1301]: May want to change the comments after this to indicate the proper byte size due to these change, just have to add 2 bytes to everything after this.

**Review by mackal — APPROVED** [review]

**Review by Kinglykrab — APPROVED** [review]

File: common/spdat.h (hunk 1 above)
Comment (mackal) [reply common/spdat.h:1301]: Comments are just a running column count of the CSV spells file.

Comment (Kinglykrab) [reply common/spdat.h:1301]: Ah, didn't even read them, some places we have bytes.

Comment (Aeadoin) [reply common/spdat.h:1301]: Good to know, if I do encounter that! @Kinglykrab

**Pull request merged**